lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs

```diff
@@ -558,6 +566,41 @@ impl FilesystemBackend {
         Ok(())
     }
 
+    /// Helper function for sending a [`tdp::SharedDirectoryReadRequest`] to the browser
+    /// and handling the [`tdp::SharedDirectoryReadResponse`] that is received in response.
+    fn tdp_sd_read(&mut self, rdp_req: efs::DeviceReadRequest) -> PduResult<()> {
+        if let Some(file) = self.file_cache.get(rdp_req.device_io_request.file_id) {
```

Comment:
Using `match` here could be safer (now if you remove `return Ok(());` by accident it will happily execute implied `None` case)

Reply:
8371f0153264cc6015da2011f18d72826306fc99